### ISSUE TITLE
Web: 3.17 tracking issue

### ISSUE THREAD

**Issue by lguychard** [sourcegraph/sourcegraph#10739]
### Availability

Period is from **May 21st** to **June 19th**. Please write the days you won't be working and the number of working days for the period.

- @lguychard: 22d
- @twop: 21d
- @attfarhan: 21d (out May 25)
- @felixfbecker: 21d (observing national holidays May 21)
- @marekweb: 22d

### Workload

@attfarhan: __6.00d__

- [ ] Explore alternatives to Reactstrap [#6443](https://github.com/sourcegraph/sourcegraph/issues/6443) __3d__ 🧶
- [x] Try enzyme for cleaner snapshot tests [#9040](https://github.com/sourcegraph/sourcegraph/issues/9040) __0.5d__ 🧶
- [x] version contexts feedback: include warning when URL changes context [#10769](https://github.com/sourcegraph/sourcegraph/issues/10769) __1d__ 
- [x] Question mark button doesn't do anything on version context popup [#10781](https://github.com/sourcegraph/sourcegraph/issues/10781) __0.5d__ 🐛
- [x] Hide 'Select context' placeholder label after first use [#10877](https://github.com/sourcegraph/sourcegraph/issues/10877) __1d__ 

@felixfbecker: __5.00d__

- [ ] Incorrect z-indexes in UI [#8380](https://github.com/sourcegraph/sourcegraph/issues/8380) __1d__ 🐛
- [x] Add visual indication of tooltip pinning [#4938](https://github.com/sourcegraph/sourcegraph/issues/4938) __4d__ 
  - [x] Overhaul tooltip close button & badge styles [#10956](https://github.com/sourcegraph/sourcegraph/pull/10956) :shipit:
- [x] Prototype code insights to track technology use [#10828](https://github.com/sourcegraph/sourcegraph/issues/10828) 
- [x] Basic code intel icon obscures type signature [#10459](https://github.com/sourcegraph/sourcegraph/issues/10459) 
  - [x] Overhaul tooltip close button & badge styles [#10956](https://github.com/sourcegraph/sourcegraph/pull/10956) :shipit:

@lguychard

- [ ] ~Improve privacy messaging before requesting addition of repository~ [#11044](https://github.com/sourcegraph/sourcegraph/issues/11044) __0.5d__ 
- [ ] ~sourcegraph/security-issues~ [#72](https://github.com/sourcegraph/security-issues/issues/72) 

@marekweb: __2.00d__

- [x] Refactor browser extension FileInfo to disallow invalid states at the type level [#3107](https://github.com/sourcegraph/sourcegraph/issues/3107) __2d__ 🧶
  - [x] Refactor FileInfo in browser extension [#10779](https://github.com/sourcegraph/sourcegraph/pull/10779) :shipit:
- [ ] ~Surface availability of the browser extension in a more persistent way~ [#10409](https://github.com/sourcegraph/sourcegraph/issues/10409) 
- [ ] ~Explore inline install of the browser extension from Sourcegraph.com~ [#10589](https://github.com/sourcegraph/sourcegraph/issues/10589) 

@twop: __16.00d__

- [x] Basic code intel info button is wrong size and color [#9065](https://github.com/sourcegraph/sourcegraph/issues/9065) 🐛
- [ ] Placeholder (maybe split later): migrate the rest of the areas to the new model according to RFC 155. [#10750](https://github.com/sourcegraph/sourcegraph/issues/10750) __8d__ 
- [ ] Migrate ExtExtensions API to a new model described in RFC 155 [#10749](https://github.com/sourcegraph/sourcegraph/issues/10749) __3d__ 
- [ ] Migrate several services/areas of extension API to a new model described in RFC 155 (tentatively Settings, Workspaces e.t.c) [#10748](https://github.com/sourcegraph/sourcegraph/issues/10748) __5d__ 
- [x] migrating commands according to RFC 155 [#11033](https://github.com/sourcegraph/sourcegraph/pull/11033) :shipit:
- [x] support predefined icons for badges (backward compat) [#10517](https://github.com/sourcegraph/sourcegraph/pull/10517) :shipit:

**Comment by attfarhan** [comment]
## W18 summary
- Shipped version contexts frontend, version contexts extension API updates

## W19 plan
- 1 follow-up call, coding challenge
- 3.18 planning
- Add warning when version contexts are changed by the URL (https://github.com/sourcegraph/sourcegraph/issues/10769)
- Move more components to Reach UI (https://github.com/sourcegraph/sourcegraph/issues/6443)

**Comment by lguychard** [comment]
## W18 Summary

- Numerous intro calls
- Improved permissions warning in the browser extension: https://github.com/sourcegraph/sourcegraph/pull/10588
- Fixed remaining smart search issues, and turned feature flag on by default: https://github.com/sourcegraph/sourcegraph/pull/10636, https://github.com/sourcegraph/sourcegraph/pull/10647, https://github.com/sourcegraph/sourcegraph/pull/10648
- Spent most of Friday fixing e2e tests so that we could build release candidates for 3.16
- Hiring

## W19 Plan

- Finish work on integration tests
- 3.17 planning

**Comment by felixfbecker** [comment]
## W18 Summary

- Improve markdown sanitization for code insights #10551
- Fix our LSIF setup #10577
- Fix pattern type redirection #10609
- Add support for charts to the extension API #10550
- Take percy snapshots of storybooks #10622
- Make NotificationItem story not flaky #10724
- Use release branch as Percy target #10726
- Prototype of resizable dashboards insights #10722
- Add test for renderMarkdown() #10727
- Reduce bundle size by 28% #10745
- Fix CI flakiness: Ignore changes to ThirdPartyLicenses in go generate check #10743
- Discuss RFC 155 with Simon
- Dependency upgrades

## W19 Plan

- Implement code insights to have a great demo
- Investigate persistent resizable grid
- 3.17 planning

**Comment by twop** [comment]
## W18 Summary
-  planning/investigating [RFC 155](https://docs.google.com/document/d/1ikrUNVe3YVbR-JpegxhjrFdmRkTGzTLcOMkKHnOyjuE/edit#)
- synced up with @felixfbecker and @lguychard about it (btw, tnx a lot!)
- almost done with https://github.com/sourcegraph/sourcegraph/pull/10517

## W19 Plan
- finish https://github.com/sourcegraph/sourcegraph/pull/10517
- migrate several pieces of functionality to the new Architecture described in RFC 155 (Settings and hopefully some others)

**Comment by lguychard** [comment]
# W19 Summary
- Enabled tracking of code host integration usage on sourcegraph.com https://github.com/sourcegraph/sourcegraph/pull/10782
- Spent most of my time doing code reviews, various discussions/syncs and hiring work (intro calls, interviews, pipeline follow-ups).

# W20 Plan
- A big hiring week again with 10 intro calls.
- Finish work on integration tests

**Comment by marekweb** [comment]
## W19 Summary

- Onboarded into codebase for browser extension and code host integrations, with lots of help from @loic and @felixfbecker
- Worked on a draft PR #10779 to fix #3107 

## W20 Plan

- Work towards finishing #10779 
- Explore solutions for browser extension install improvements
  - Improve showing extension availability #10409
  - Explore inline install #10589

**Comment by twop** [comment]
## W19 Summary
- finished https://github.com/sourcegraph/sourcegraph/pull/10517
- Migrated settings ( #10874) and WIP workspaces. (https://github.com/sourcegraph/sourcegraph/pull/10944)

## W20 Plan
- continue working on migration https://github.com/sourcegraph/sourcegraph/issues/10748

**Comment by felixfbecker** [comment]
## W19 Summary
- Follow-up fixes for Firefox to bundle size reduction #10785, #10789
- Customer issue: Treat 403s as auth errors #10777
- Overhaul close button and badge styling (fix overlap and pinning state confusion) #10956
- Code insights: Propose LSIF diagnostics API for code insights #10827
- Code insights: Propose search feature for code insights #10820
- Help @twop and @marekweb
- Thursday holiday

## W20 Plan

- Find out what insights we need to build. Work will be for that goal, but potentially:
- Technology usage insights #10828
- Investigate global insights #10829
- Investigate persistent dashboards
- Development metrics
- Represent our OKRs as code insights?

**Comment by lguychard** [comment]
## W20 Update

- Spent a lot of time in intro calls, code review and other syncs.
- Battling a little bit to make Polly do what I want for integration tests - their Puppeteer adapter is pretty much broken so I had to write a new one.
- Started flattening OptionsMenu code -- PR incoming soon.
- Wrote down some thoughts on making changes to how our Firefox browser extension executes Sourcegraph extensions in [RFC 173](https://docs.google.com/document/d/1YN9xdRD7RhAkCrR2RYbIQpYrC_wrVNteuzp_nCguuoI/edit). We might have to make a prioritization call on this soon.

**Comment by felixfbecker** [comment]
## W20 Summary

- Make browser extension folder structure clearer https://github.com/sourcegraph/sourcegraph/pull/11026
- Adopt eslint-plugin-unicorn https://github.com/sourcegraph/eslint-config/pull/91
- Work on testing and styling documentation
- Discuss design process with Rob
- Start working on storybook additions for global styles for Rob
- Review and discuss Marek's and Simon's awesome PRs

## W21 Plan

- Finish testing and styling documentation
- Extend storybook with global styles
- Adapt ESLint prevent-abbreviations rule
- Draft of code insights marketing page
- Start on code insights work mentioned last week:
  > Find out what insights we need to build. Work will be for that goal, but potentially:
  > - Technology usage insights #10828
  > - Investigate global insights #10829
  > - Investigate persistent dashboards
  > - Development metrics
  > - Represent our OKRs as code insights?

**Comment by twop** [comment]
## W20 Summary
- Badge icons https://github.com/sourcegraph/sourcegraph/issues/9065
  - sent a PR that fixing the issue for code-intel https://github.com/sourcegraph/code-intel-extensions/pull/395
  - published new "sourcegraph" package to npm 
- investigated moving commands registry to the worker (will not move)
- finished migrating workspaces https://github.com/sourcegraph/sourcegraph/pull/10944

## W21 Plan
- [x] finished migrating commands https://github.com/sourcegraph/sourcegraph/pull/11033
- Write RFC/capture ideas around CSS technologies and workflow
- Migrate ExtExtensions API to a new model described in RFC 155 #10749 
- feedback on [RFC 175: Design to Dev workflow](https://docs.google.com/document/d/190Ong7kRkNFt6nSDeVv-OUg7AlQdNYW91fxrSPWQ8dk/edit#)
- potentially work on race condition propagating settings to extension https://github.com/sourcegraph/sourcegraph/issues/11235

**Comment by marekweb** [comment]
## W20 Summary

- Completed most work on #10779 and worked on testing
- Started on #10409 showing extension availability 
- Explored inline install #10589 
- Collected documentation notes for code views in code hosts

## W21 Plan

- Complete testing of #10779
- Get familiar with issues unblocked by #10779 which are tracked in #4377
- Work on #10409 show extension availability
- Investigate and test GitHub redesign #11011
- Explore CSS scoping #10595

**Comment by felixfbecker** [comment]
## W21 Summary

- Dependency upgrade week!
- Finished testing documentation #1036
- Extended storybook with global styles for Rob #11158
- Adopt ESLint prevent-abbreviations rule
- Started on code insights marketing page
- Discussions on extension host
- Discussions of extensions

## W22 

- Work on and finish PR for code insights marketing page
- Code insights work specifically to go to https://app.hubspot.com/contacts/2762526/company/407948923 and get feedback:  Technology usage insights #10828
- Build demo on code-intel's new diagnostics support #11233 
- Other items:
  > - Investigate global insights #10829
  > - Investigate persistent dashboards
  > - Development metrics
  > - Represent our OKRs as code insights?

**Comment by marekweb** [comment]
## W21 Summary
- Finished testing PR #10779 
- Investigated issues unblocked by #10779 which are tracked in #4377
- Started on #10409 show extension availability

## W22 Plan
- Priority: Work on inlining code intel extensions for Firefox #11369

**Comment by felixfbecker** [comment]
# W22 Summary

- Get testing guide merged https://github.com/sourcegraph/about/pull/1036
- Added global insights #11433
- Implemented technology usage tracking extension https://github.com/sourcegraph/sourcegraph/issues/10828
- Demoed to https://app.hubspot.com/contacts/2762526/company/407948923 and gathered feedback
- Started uploading ESLint warnings as LSIF diagnostics #11469
- Discussions on extension host

# W23 Plan

- Respond to Simon's thoughts on testing https://github.com/orgs/sourcegraph/teams/web/discussions/1
- Build ESLint diagnostic insights
- Plan 3.18 work

**Comment by twop** [comment]
## W22 Summary
- wrote my thoughts on testing here https://github.com/orgs/sourcegraph/teams/web/discussions/1
- Migrated Query transformer api #10748. ready for review (not checked in yet)
- Did a research on design -> dev workflow tooling [RFC 175](https://docs.google.com/document/d/190Ong7kRkNFt6nSDeVv-OUg7AlQdNYW91fxrSPWQ8dk/edit). 
- Discussed/brainstorm next steps for firefox add-on from extensions point of view with @marekweb 


## W23 Plan
- check in Query transformer api #10748
- Migrate language provider(s) to according to RFC 155 (#10750 ). Will postpone working on extension activation flow until @marekweb done working on Firefox add-on (#10749 ).
- (?) Maybe write my thoughts as RFC related to dealing with CSS.

**Comment by uwedeportivo** [comment]
Dear all,

This is your release captain speaking. 🚂🚂🚂

Branch cut for the **3.17 release is scheduled for tomorrow**.

Is this issue / PR going to make it in time? Please change the milestone accordingly.
When in doubt, reach out!

Thank you
